dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -345,13 +358,29 @@ def _update_categories(
         :type df_series: pandas.DataFrame
         :return: None
         """
-        category_count = df_series.value_counts(dropna=False).to_dict()
-        self._categories = utils.add_nested_dictionaries(
-            self._categories, category_count
-        )
-        self._update_stop_condition(df_series)
-        if self._stop_condition_is_met:
-            self._categories = {}
+        if self.cms is not None:
+            if self._top_k_categories is None:
+                raise ValueError("when using CMS, top_k must be an integer")
+            category_count = defaultdict(int)
+            for i in df_series:
+                self.cms.update(i)
+                # approximate heavy-hitters
+                if self.cms.get_estimate(i) >= int(
+                    self.sample_size / self._top_k_categories
```

Comment:
thoughts on creating a new var self._heavy_hitters_threshold instead of misusing top_k_categories here? In the CMS setting we are using top_k_categories as a minimum count for us to track a category, this doesn't imply we'll end up with k categories...